### PR TITLE
UI tests

### PR DESCRIPTION
**Pull request by VyacheslavHalpukov**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2021-2022 Intel Corporation
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_119_change_status_task.js:1]:
```suggestion
// Copyright (C) 2022 CVAT.ai Corporation
```

File: tests/cypress/integration/actions_users/issue_1810_login_logout.js [hunk 2]
```diff
@@ -20,6 +20,39 @@ context('When clicking on the Logout button, get the user session closed.', () =
...
+            }
+        }
+        const userMessage = 'Please specify a email or username';
+        const passwordMessage = 'Please specify a password';
+        if (userName) {
+            cy.get(locators.userField).type(userName);
+            cy.get(locators.submitButton).click();
+            cy.contains(passwordMessage).should('exist');
+        }
+        else if (password) {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/issue_1810_login_logout.js:40]:
```suggestion
        } else if (password) {
```

File: tests/cypress/integration/actions_users/issue_1810_login_logout.js [hunk 3]
```diff
@@ -20,6 +20,39 @@ context('When clicking on the Logout button, get the user session closed.', () =
...
+            cy.get(locators.passwordField).type(password);
+            cy.get(locators.submitButton).click();
+            cy.contains(userMessage).should('exist');
+        }
+        else {
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/issue_1810_login_logout.js:45]:
```suggestion
        } else {
```

File: tests/cypress/integration/actions_users/issue_1810_login_logout.js [hunk 4]
```diff
@@ -97,5 +130,25 @@ context('When clicking on the Logout button, get the user session closed.', () =
...
+            loginWithoutCredentials(false, Cypress.env('password'));
+        });
+
+        it('Incorrect user and empty password', () => {
+            loginWithoutCredentials('randomUser123', false);
+        });
+
+        it('Empty user and incorrect password', () => {
+            loginWithoutCredentials(false, 'randomPassword123');
+        });
```
Comment (bsekachev) [inline tests/cypress/integration/actions_users/issue_1810_login_logout.js:152]: Can we parametrize call of ``loginWithoutCredentials`` in a ``for`` loop? Like 
```
for (const [username, password] of [[user1, pass1], [user2, pass2] ,...) { 
    loginWithoutCredentials(username, password)
}
```

I personally do not think we need such number of negative tests, it is like we are trying to test the framework we are using on UI

File: tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2021-2022 Intel Corporation
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:1]:
```suggestion
// Copyright (C) 2022 CVAT.ai Corporation
```

File: tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            cy.openTask(taskName);
+            cy.setJobStage(0, 'acceptance');
+            cy.goToTaskList();
+            checkCompletedJobs(1, 3);
+            cy.openTask(taskName);
+            cy.setJobStage(1, 'acceptance');
+            cy.goToTaskList();
+            checkCompletedJobs(2, 3);
+            cy.openTask(taskName);
+            cy.setJobStage(2, 'acceptance');
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:69]: To simplify I would suggest this:

```suggestion
            cy.setJobStage(0, 'acceptance');
            cy.setJobStage(1, 'acceptance');
            cy.setJobStage(2, 'acceptance');
```

File: tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            textDefaultValue,
+            archiveName,
+            false,
+            advancedConfigurationParams,
+        );
+    });
+
+    after(() => {
+        cy.goToTaskList();
+        cy.deleteTask(taskName);
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:47]: Better to delete task in headless mode, like a kind of routine work. We do not test deleting a task here

Comment (bsekachev) [reply tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:47]: You can see how it can be done in some other tests, like:
```
cy.getAuthKey().then((response) => {
            const authKey = response.body.key;
            cy.request({
                method: 'DELETE',
                url: `/api/tasks/${taskID}`,
                headers: {
                    Authorization: `Token ${authKey}`,
                },
            });
        });
```

We could also have a command like ``deleteTaskHeadless`` with this code

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    const posY = 10;
+    const color = 'gray';
+    const archiveName = `${imageFileName}.zip`;
+    const archivePath = `cypress/fixtures/${archiveName}`;
+    const imagesFolder = `cypress/fixtures/${imageFileName}`;
+    const directoryToArchive = imagesFolder;
+    const advancedConfigurationParams = {
+        multiJobs: true,
+        segmentSize: 5,
+    };
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_119_change_status_task.js:27]: Consider code from ``cvat/tests/cypress/integration/masks/masks_basics.js`` to create a task, it works much faster, because it is being done in headless mode

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+            cy.goToTaskList();
+            checkTaskStatus('In Progress');
+        });
+
+        it('Not changed task status "In progress".', () => {
+            cy.openTask(taskName);
+            cy.setJobStage(1, 'acceptance');
+            cy.goToTaskList();
+            checkTaskStatus('In Progress');
+        });
```
Comment (bsekachev) [inline tests/cypress/integration/actions_tasks/case_119_change_status_task.js:70]: I consider this ``it`` to be extra since it does not test anything new

**Comment by bsekachev** [comment]
I see ``ESLint`` pipeline and ``CI / e2e_testing (actions_users)`` pipeline failed..

File: tests/cypress/integration/actions_users/issue_1810_login_logout.js (hunk 4 above)
Comment (VyacheslavHalpukov) [reply tests/cypress/integration/actions_users/issue_1810_login_logout.js:152]: I parameterized this test and the test with the wrong password or name in the way you suggested.

This is the entry point, all users go through it, I think it is reasonable to check all corner cases. This test takes seconds to pass.

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js (hunk 8 above)
Comment (VyacheslavHalpukov) [reply tests/cypress/integration/actions_tasks/case_119_change_status_task.js:27]: `cy.headlessCreateTask({
            labels: [{ name: labelName, attributes: [], type: 'any' }],
            name: taskName,
            // TODO: add support for multi jobs
            // segments: [{
            //     jobs: []
            // }],
            project_id: null,
            source_storage: { location: 'local' },
            target_storage: { location: 'local' },
        }, {
            server_files: serverFiles,
            image_quality: 70,
            use_zip_chunks: true,
            use_cache: true,
            sorting_method: 'lexicographical',
        }).then((response) => {
            console.log(response);
            taskId = response.taskID;
        }).then(() => {
            cy.visit(`/tasks/${taskId}`);
        });`

Using the option you suggested, the method creates only one job attached.

Can you highlight which object to send to create a Task with multijob ?

File: tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js (hunk 7 above)
Comment (VyacheslavHalpukov) [reply tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:47]: I will change all the tests I created to this method

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js (hunk 9 above)
Comment (VyacheslavHalpukov) [reply tests/cypress/integration/actions_tasks/case_119_change_status_task.js:70]: It's good practice to run tasks across the flow. But I can also make it shorter.

File: tests/cypress/integration/actions_tasks/case_119_change_status_task.js (hunk 8 above)
Comment (bsekachev) [reply tests/cypress/integration/actions_tasks/case_119_change_status_task.js:27]: Need to specify ``segment_size: 1``` after ``sorting_method`` for example.

File: tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js (hunk 6 above)
Comment (VyacheslavHalpukov) [reply tests/cypress/integration/actions_tasks/case_120_change_completed_jobs_task.js:69]: Made the test more universal.

I still think that in UI tests you should go through the interface and check the status of elements. In this test it doesn't take much time.
